app/mailers/group_smtp_mailer.rb

```diff
@@ -38,29 +33,32 @@ def send_mail(from_group, to_address, post)
       user_name = post.user.name unless post.user.name.blank?
     end
 
-    build_email(to_address,
+    group_name = from_group.full_name.presence || from_group.name
+    build_email(
+      to_address,
       message: post.raw,
       url: post.url(without_slug: SiteSetting.private_email?),
       post_id: post.id,
       topic_id: post.topic_id,
       context: context(context_posts),
       username: post.user.username,
-      group_name: from_group.name,
+      group_name: group_name,
       allow_reply_by_email: true,
       only_reply_by_email: true,
-      use_from_address_for_reply_to: SiteSetting.enable_imap && from_group.imap_enabled?,
+      use_from_address_for_reply_to: SiteSetting.enable_smtp && from_group.smtp_enabled?,
       private_reply: post.topic.private_message?,
       participants: participants(post),
       include_respond_instructions: true,
       template: 'user_notifications.user_posted_pm',
       use_topic_title_subject: true,
-      topic_title: incoming_email&.subject || post.topic.title,
+      topic_title: post.topic.title,
```

Comment:
Hmm good point, this may be (possibly?) causing some issues with the sent emails not showing email. Will put this back, and just use the incoming email from the OP.